src/adapter/src/coord/sequencer.rs

```diff
@@ -3110,6 +3115,84 @@ impl<S: Append + 'static> Coordinator<S> {
         Ok(ExecuteResponse::AlteredObject(ObjectType::Secret))
     }
 
+    async fn sequence_alter_source(
+        &mut self,
+        session: &Session,
+        AlterSourcePlan { id, size, remote }: AlterSourcePlan,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        use mz_sql::ast::Value;
+        use AlterSourceItem::*;
+        use CreateSourceOptionName::*;
+
+        let entry = self.catalog.get_entry(&id);
+        let old_source = match entry.item() {
+            CatalogItem::Source(source) => source.clone(),
+            other => coord_bail!("ALTER SOURCE entry was not a source: {}", other.typ()),
+        };
+
+        // Since the catalog serializes the items using only their creation statement
+        // and context, we need to parse and rewrite the with options in that statement.
+        // (And then make any other changes to the source definition to match.)
+        let mut stmt = mz_sql::parse::parse(&old_source.create_sql)
+            .unwrap()
+            .into_element();
+
+        let create_stmt = match &mut stmt {
+            Statement::CreateSource(s) => s,
+            _ => coord_bail!("source {id} was not created with a CREATE SOURCE statement"),
+        };
+
+        let new_config = match (&old_source.host_config, size, remote) {
+            (_, Set(_), Set(_)) => coord_bail!("Can't set both SIZE and REMOTE on source"),
+            (_, Set(size), _) => Some(plan::StorageHostConfig::Managed { size }),
+            (_, _, Set(addr)) => Some(plan::StorageHostConfig::Remote { addr }),
+            (StorageHostConfig::Remote { .. }, _, Reset)
+            | (StorageHostConfig::Managed { .. }, Reset, _) => {
+                Some(plan::StorageHostConfig::Undefined)
+            }
+            (_, _, _) => None,
+        };
+
+        if let Some(config) = new_config {
+            let host_config = self.catalog.resolve_storage_host_config(config)?;
+
+            const HOST_CONFIG_OPTIONS: &[CreateSourceOptionName] = &[Size, Remote];
+            let (name, value) = match &host_config {
+                StorageHostConfig::Managed { size, .. } => (Size, size.clone()),
+                StorageHostConfig::Remote { addr } => (Remote, addr.clone()),
+            };
+            create_stmt
+                .with_options
+                .retain(|x| !HOST_CONFIG_OPTIONS.contains(&x.name));
+            create_stmt.with_options.push(CreateSourceOption {
```

Comment:
I think that this will translate a`RESET SIZE` into a SQL statement that verbatim specifies the current size default as opposed to leaving it unspecified so that each time the alter source is planned we pick the current default. I think you need to write the same logic but directly on `config` and handle the `Undefined` case by not pushing any option

Reply:
Yeah, just saw the discussion in Slack -- I'll make the change!